src/WebJobs.Script/Workers/Rpc/RpcWorkerChannel.cs

```diff
@@ -136,12 +139,21 @@ internal void SendWorkerInitRequest(RpcEvent startEvent)
                 .Take(1)
                 .Subscribe(WorkerInitResponse, HandleWorkerInitError);
 
+            var initRequest = new WorkerInitRequest()
+            {
+                HostVersion = ScriptHost.Version,
+            };
+
+            // Run as Functions Host V2 compatable
+            if (_environment.IsV2CompatibilityMode())
+            {
+                _workerChannelLogger.LogDebug("Worker and host running in V2 compatability mode");
+                initRequest.Capabilities.Add(RpcWorkerConstants.V2Compatable, "true");
+            }
```

Comment:
Do we re-send init requests during specialization? And the worker can handle that if the V2 Compat changes then?

Reply:
Great catch, tracking here: #5323